cmd/containerd-shim-runhcs-v1/delete.go

```diff
@@ -126,28 +123,8 @@ The delete command will be executed in the container's bundle as its cwd.
 			fmt.Fprintf(os.Stderr, "failed to delete user %q: %v", username, err)
 		}
 
-		// cleanup the layers mounted for the container. We currently only handle cleanup of CimFS
-		// layers here.  First n-1 values should be the image layerFolders (topmost layer being at
-		// index 0) and the last entry should be the scratch layer
-		var layerFolders []string
-		f, err := os.Open(filepath.Join(bundleFlag, layersFile))
-		if err != nil {
-			if !errors.Is(err, os.ErrNotExist) {
-				fmt.Fprintf(os.Stderr, "open layers file: %s", err)
-			}
-		} else {
-			defer f.Close()
-			if err = json.NewDecoder(f).Decode(&layerFolders); err != nil {
-				fmt.Fprintf(os.Stderr, "decode layers json: %s", err)
-			}
-		}
-		if err == nil && cimlayer.IsCimLayer(layerFolders[0]) {
-			scratchLayerFolderPath := layerFolders[len(layerFolders)-1]
-			err = layers.ReleaseCimFSHostLayers(ctx, scratchLayerFolderPath, idFlag)
-			if err != nil {
-				fmt.Fprintf(os.Stderr, "cleanup container %q mounts: %s", idFlag, err)
-			}
-		}
+		// TODO(ambarve):
+		// correctly handle cleanup of cimfs layers in case of shim process crash here.
```

Comment:
What's the current thinking here? Are we still evaluating with upstream if the new deletion behavior should be changed?

Reply:
The old behavior of container was actually due to a bug, that means with 2.0 onwards we need to persist layer data for each container started within a shim and properly clean it up during `shim delete` call. However, that's a medium/large change so it needs to be handled in a separate PR.